index.js

```diff
@@ -1,131 +1,140 @@
+            });
+    }
 
-  async generateSignal(response,upColor,downColor,upEffect,downEffect){
-    //if the answer is json response with notification array
-    if(response.notifications){
-      let color=upColor;
-      let effect=upEffect;
-      let number=0;
-      for (let notification of response.notifications) {
-        let isRead = notification.read;
-        let notifID = notification.id;
+    /**
+     * Send a q signal according to the response from API request
+     *
+     * @param {*} response : json response from API
+     * @param {*} warnerColor : key color 
+     * @param {*} warnerEffect :
+     */
+    async generateSignal(response, warnerColor, warnerEffect,host) {
+        // if notifications exist in response
+        if (response.notifications) {
+            let signal = null;
+            let notificationNumber = 0;
+            for (let notification of response.notifications) {
+                let isRead = notification.read;
 
-        logger.info(`Notification ${notifID} is ${isRead ? "read" : " unread"}`);
-        
-        if (!isRead) {
-          effect = downEffect;
-          number++;
-          color = downColor;
-          alerts.push(notifID);
-        }
-      }
+                // check if the notification is read or not
+                if (!isRead) {
+                    notificationNumber++;
+                }
+            }
 
-      logger.info("you have " + number + " notifications" + " unread");
+            // check if there are notifications
+            if (notificationNumber != 0) {
+                let message =
+                    notificationNumber == 1
+                        ? '1 unread notification on ' + host
+                        : notificationNumber +
+                          ' notifications unread on ' +
+                          host;
 
-      //look if we got at least one notification unread
-      if (number != 0) {
-        let signal = new q.Signal({
-          points: [[new q.Point(color, effect)]],
-          name: this.config.rootURL,
-          message:"You have "+number+" unread notifications with id's :"+alerts.join(", "),
-          link: {
-            url: this.config.rootURL,
-            label: "Open discourse web site",
-          },
-        });
-        return signal;
-      }
-      else {
-        let signal = new q.Signal({
-          points: [[new q.Point(color, effect)]],
-          name: this.config.rootURL,
-          message: "You have no unread notifications",
-          link: {
-            url: this.config.rootURL,
-            label: "Open discourse web site",
-          },
-        });
-        return signal;
-      }
-    }
-    //if the answer is an error
-    else{
-      return response;
+                signal = new q.Signal({
+                    points: [[new q.Point(warnerColor, warnerEffect)]],
+                    name: 'Discourse',
+                    message: message,
+                    link: {
+                        url:
```

Comment:
Could use \`${host}/u/${this.config.username}/notifications?filter=unread\` easier to read